### DISCUSSION TITLE
Migrate ManagedExecutorService to Quarkus

### DISCUSSION THREAD

**Discussion by towbi** [quarkusio/quarkus#23253] [Q&A]
Hi there,

I'm in the process of migrating an existing TomEE (JavaEE) application to Quarkus. In this application we make frequent use of ManagedExecutorServices that get injected using the @Resource annotation like this:

```
    @ApplicationScoped
    public class MyService {

        @Resource(name = "MyManagedExecutorService")
        private ManagedExecutorService managedExecutorService;
	...
	    managedExecutorService.submit(() -> {
	        ...
	    });
    ...
```

The named resource MyManagedExecutorService is defined in WEB-INF/resources.xml like this:

```
    <resources>
        <Resource id="MyManagedExecutorService" type="ManagedExecutorService">
            Core = 100
            Max = 100
            KeepAlive = 5 s
            Queue = 1000
        </Resource>
    ...
```

I researched quite a bit bit I didn't find a way to migrate these parts to Quarkus without major rewrites of the application.

Is there a straightforward way to migrate these parts to Quarkus-land?

Cheers,
towbi

**Comment by fwippe** [comment]
You could use the Quarkus default (configurable) executor and just inject a `ManagedExecutor`:

```java
@Inject ManagedExecutor executor;
```

**Reply by lvian** [reply]
This solved the issue for my case.

Just had to add `implementation 'io.quarkus:quarkus-smallrye-context-propagation-spi:2.14.1.Final'` to our build.gradle

**Comment by BhumikaRupera** [comment]
> Hi there,
> 
> I'm in the process of migrating an existing TomEE (JavaEE) application to Quarkus. In this application we make frequent use of ManagedExecutorServices that get injected using the @resource annotation like this:
> 
> ```
>     @ApplicationScoped
>     public class MyService {
> 
>         @Resource(name = "MyManagedExecutorService")
>         private ManagedExecutorService managedExecutorService;
> 	...
> 	    managedExecutorService.submit(() -> {
> 	        ...
> 	    });
>     ...
> ```
> 
> The named resource MyManagedExecutorService is defined in WEB-INF/resources.xml like this:
> 
> ```
>     <resources>
>         <Resource id="MyManagedExecutorService" type="ManagedExecutorService">
>             Core = 100
>             Max = 100
>             KeepAlive = 5 s
>             Queue = 1000
>         </Resource>
>     ...
> ```
> 
> I researched quite a bit bit I didn't find a way to migrate these parts to Quarkus without major rewrites of the application.
> 
> Is there a straightforward way to migrate these parts to Quarkus-land?
> 
> Cheers, towbi

What were the rewrites that you had to do for making this work? I am facing a similar issue.
